Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -866,24 +876,27 @@ TEST_F(SerializerTest, FailOnInvalidInputBytes) {
 }
 
 TEST_F(SerializerTest, EncodesKey) {
-  EXPECT_EQ("projects/p/databases/d/documents", serializer.EncodeKey(Key("")));
+  EXPECT_EQ("projects/p/databases/d/documents",
+            FromBytes(serializer.EncodeKey(Key(""))));
   EXPECT_EQ("projects/p/databases/d/documents/one/two/three/four",
-            serializer.EncodeKey(Key("one/two/three/four")));
+            FromBytes(serializer.EncodeKey(Key("one/two/three/four"))));
 }
 
 TEST_F(SerializerTest, DecodesKey) {
   Reader reader(nullptr, 0);
-  EXPECT_EQ(Key(""),
-            serializer.DecodeKey(&reader, "projects/p/databases/d/documents"));
+  EXPECT_EQ(Key(""), serializer.DecodeKey(
+                         &reader, ToBytes("projects/p/databases/d/documents")));
   EXPECT_EQ(
       Key("one/two/three/four"),
       serializer.DecodeKey(
-          &reader, "projects/p/databases/d/documents/one/two/three/four"));
+          &reader,
+          ToBytes("projects/p/databases/d/documents/one/two/three/four")));
```

Comment:
Similarly every usage here leaks.

Another possibility would be make `ToBytes` and `FromBytes` members and track each of these temporary strings so that they can be freed during test teardown.

Reply:
Make `ToBytes` return a `ByteString`. It makes the tests _slightly_ more verbose (because of the call to `get`).

Is there a particular reason `Serializer` mostly deals with the raw `pb_bytes_array_t` and not `ByteString`?